libtransact/src/execution/adapter/static_adapter.rs

```diff
@@ -0,0 +1,628 @@
+
+        let (send, recv) = std::sync::mpsc::channel();
+        static_adapter.execute(
+            txn_pair,
+            context_id.clone(),
+            Box::new(move |res| {
+                send.send(res).expect("Unable to send result");
+            }),
+        );
+        let result = recv.recv().unwrap();
+        assert_eq!(
+            ExecutionTaskCompletionNotification::Invalid(
+                context_id,
+                InvalidTransactionResult {
+                    transaction_id: txn_id,
+                    error_message: "Test Fail Succeeded".into(),
+                    error_data: vec![],
+                }
+            ),
+            result.unwrap()
+        );
+    }
+
+    fn make_transaction(commands: &[Command]) -> TransactionPair {
+        let signer = HashSigner::new();
+        TransactionBuilder::new()
+            .with_batcher_public_key(vec![0u8, 0u8, 0u8, 0u8])
+            .with_family_name(COMMAND_FAMILY_NAME.to_owned())
+            .with_family_version(COMMAND_VERSION.to_owned())
+            .with_inputs(
+                commands
+                    .iter()
+                    .map(|cmd| match cmd {
+                        Command::Set { address, .. } | Command::Delete { address } => {
+                            Some(address.as_bytes().to_vec())
+                        }
+                        _ => None,
+                    })
+                    .filter(Option::is_some)
+                    .map(Option::unwrap)
+                    .collect(),
+            )
+            .with_outputs(
+                commands
+                    .iter()
+                    .map(|cmd| match cmd {
+                        Command::Set { address, .. } | Command::Delete { address } => {
+                            Some(address.as_bytes().to_vec())
+                        }
+                        _ => None,
+                    })
+                    .filter(Option::is_some)
+                    .map(Option::unwrap)
+                    .collect(),
+            )
+            .with_payload_hash_method(HashMethod::SHA512)
+            .with_payload(
+                commands
+                    .iter()
+                    .map(ToString::to_string)
+                    .collect::<Vec<_>>()
+                    .join("|")
+                    .into_bytes(),
+            )
+            .build_pair(&signer)
+            .unwrap()
+    }
+
+    /// A basic Command Transaction Family
+
+    const COMMAND_FAMILY_NAME: &str = "command";
+    const COMMAND_VERSION: &str = "0.1";
+
+    #[derive(Default)]
+    struct CommandTransactionHandler {
+        versions: Vec<String>,
+    }
+
+    impl CommandTransactionHandler {
+        pub fn new() -> Self {
+            CommandTransactionHandler {
+                versions: vec![COMMAND_VERSION.to_owned()],
+            }
+        }
+    }
+
+    impl TransactionHandler<String, Vec<u8>> for CommandTransactionHandler {
```

Comment:
Should we create a top-level handlers module? We could enable this for tests and/or a feature.

Reply:
Sounds like something for a future PR.